app/src/main/java/io/intrepid/skeleton/base/BasePresenterImpl.java

```diff
@@ -0,0 +1,71 @@
+package io.intrepid.skeleton.base;
+
+import android.support.annotation.NonNull;
+
+import io.intrepid.skeleton.rest.RestApi;
+import io.intrepid.skeleton.settings.UserSettings;
+import rx.Observable;
+import rx.Scheduler;
+import rx.subscriptions.CompositeSubscription;
+
+public abstract class BasePresenterImpl<T extends BaseView> implements BasePresenter<T> {
+
+    protected final CompositeSubscription subscriptions = new CompositeSubscription();
+
+    protected T view;
+    @NonNull
+    protected final Scheduler ioScheduler;
+    @NonNull
+    protected final Scheduler uiScheduler;
+    @NonNull
+    protected final UserSettings userSettings;
+    @NonNull
+    protected final RestApi restApi;
+
+    public BasePresenterImpl(@NonNull T view, @NonNull PresenterConfiguration configuration) {
+        this.view = view;
```

Comment:
Neither, concrete presenters should end with Presenter, while interfaces should be in the form of XContract.Preseneter (see the Example1Contract). I should probably rename BasePresenter to BaseContract.Presenter, and BasePresenterImp to BasePresenter to keep things consistent